js_modules/dagster-ui/packages/ui-core/src/asset-graph/AssetGraphExplorerSidebar.tsx

```diff
@@ -99,24 +121,98 @@ export const AssetGraphExplorerSidebar = React.memo(
+              groupName: string;
+              assets: string[];
+            }
+          >;
+        }
+      > = {};
+      Object.entries(assetGraphData.nodes).forEach(([id, node]) => {
+        const locationName = node.definition.repository.location.name;
+        const repositoryName = node.definition.repository.name;
+        const groupName = node.definition.groupName || 'default';
+        const codeLocation = buildRepoPathForHuman(repositoryName, locationName);
+        codeLocationNodes[codeLocation] = codeLocationNodes[codeLocation] || {
+          locationName: codeLocation,
+          groups: {},
+        };
+        codeLocationNodes[codeLocation]!.groups[groupName] = codeLocationNodes[codeLocation]!
+          .groups[groupName] || {
+          groupName,
+          assets: [],
+        };
+        codeLocationNodes[codeLocation]!.groups[groupName]!.assets.push(id);
+      });
+      Object.entries(codeLocationNodes).forEach(([locationName, locationNode]) => {
+        folderNodes.push({locationName, id: locationName, level: 1});
+        if (openNodes.has(locationName)) {
+          Object.entries(locationNode.groups).forEach(([groupName, groupNode]) => {
+            const groupId = locationName + ':' + groupName;
+            folderNodes.push({groupName, id: groupId, level: 2});
+            if (openNodes.has(groupId)) {
+              groupNode.assets
+                .sort((a, b) => COLLATOR.compare(a, b))
+                .forEach((assetKey) => {
+                  folderNodes.push({
+                    id: assetKey,
+                    path: locationName + ':' + groupName + ':' + assetKey,
+                    level: 3,
+                  });
+                });
+            }
+          });
+        }
+      });
+
+      return folderNodes;
+    }, [assetGraphData.nodes, openNodes]);
+
+    const renderedNodes = viewType === 'tree' ? treeNodes : folderNodes;
+
     const containerRef = React.useRef<HTMLDivElement | null>(null);
 
     const rowVirtualizer = useVirtualizer({
       count: renderedNodes.length,
       getScrollElement: () => containerRef.current,
-      estimateSize: () => 38,
+      estimateSize: () => 28,
       overscan: 10,
     });
 
     const totalHeight = rowVirtualizer.getTotalSize();
     const items = rowVirtualizer.getVirtualItems();
 
     React.useLayoutEffect(() => {
-      if (lastSelectedNode && lastSelectedNode.id !== selectedNode?.id) {
+      if (lastSelectedNode) {
         setOpenNodes((prevOpenNodes) => {
+          if (viewType === 'folder') {
+            const nextOpenNodes = new Set(prevOpenNodes);
+            const assetNode = assetGraphData.nodes[lastSelectedNode.id];
+            if (assetNode) {
+              const locationName = buildRepoPathForHuman(
+                assetNode.definition.repository.name,
+                assetNode.definition.repository.location.name,
+              );
+              const groupName = assetNode.definition.groupName || 'default';
```

Comment:
I don't think this `|| 'default'` should be necessary -- I think the groupName was originally optional but hasn't been for a while

Reply:
Some of the nodes in the toys repo seemed not to have a groupName so I think we should keep it for back compat at least